design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+  - You have a legacy infrastructure that you want to migrate to Crossplane, but you want to test the managed resources
+  before taking ownership of the underlying resources.
+  - For an existing resource, you don’t want to give some initial configuration that might override the actual
+  configuration. You want to late-initialize all fields, including the ones that would be required otherwise.
+- Only observing some fields after the initial creation
+  - You want to create an EKS Node Group with a scaling configuration where you configured an initial desired size.
+  After the creation, you want to ignore changes in the size which is now being controlled by the cluster autoscaler.
+
+Currently, Crossplane does not have a built-in way of observing resources without taking ownership of them. There are
+two workarounds used by the community as an interim solution for this gap:
+
+- Using the provider-terraform to observe resources with the help of Terraform data sources.
+- Wrapping resources with provider-kubernetes to observe resources managed by Crossplane but as a shared object between
+multiple Compositions.
+
+In this document, we aim to introduce a solution to observe resources with Crossplane without taking ownership of them.
+This would allow users to integrate existing cloud resources with the Crossplane ecosystem without giving full ownership.
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane ecosystem.
+- Allow users to fetch data from existing resources without giving full ownership.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
+
+## Proposal
+
+To support observing resources without taking ownership, we will introduce a new spec named `managementPolicy` to the
+Managed Resources. We will also deprecate the existing `deletionPolicy` in favor of the new spec since they will be
+controlling the same behavior; that is, how should the changes on the CR affect the external cloud resource.
+
+This new policy will have the following values:
+
+- `FullControl`(Default): Crossplane will fully manage and control the external resource, including deletion when the CR
+is deleted (same as `deletionPolicy: Delete`).
+- `ObserveCreateUpdate`: Crossplane will observe and perform create and update operations on the external resource but
+will not make any deletions (same as `deletionPolicy: Orphan`).
+- `ObserveOnly`: Crossplane will only observe the external resource and will not make any changes or deletions.
+
+Please note while `ObserveCreateUpdate` may sound verbose, it accurately describes the actions that Crossplane will take
+on the external resource. This naming convention also focuses on what Crossplane will do, rather than what it won't do,
+making it more intuitive and extensible for future policy options, such as `ObserveDelete`.
```

Comment:
No, I agree.
I started with `managementPolicy: Orphan` but immediately noticed that it sounds like we will leave external resources as orphaned by not **_managing_** it. 
So, open to suggestions here, but if possible, would still like to stick to the convention of focusing on what Crossplane will do, rather than what it won't do as mentioned in the above paragraph.

Reply:
One possibility could be `managementPolicy: OrphanOnDelete` if it sounds better than `managementPolicy: ObserveCreateUpdate`.

Reply:
Ok, I am going with `OrphanOnDelete` instead: https://github.com/crossplane/crossplane/pull/3531/commits/0a183485272df16f9109778cd55e5f8d106d1d15